lib/spack/spack/binary_distribution.py

```diff
@@ -415,7 +415,7 @@ def relocate_package(workdir, allow_root):
     new_path = spack.store.layout.root
     old_path = buildinfo['buildpath']
     rel = buildinfo.get('relative_rpaths', False)
-    if new_path == old_path and not rel:
```

Comment:
@gartung I understand removing `new_path == old_path`. I'm curious if the original logic had a bug with the `and not rel` part: I'm assuming that before the placeholder logic it should have skipped relocation if the package used relative rpaths (and I don't think the old statement was doing that).

Reply:
The original probably should have been
if new_path == old_path or rel: